docs/src/main/tut/kleisli.md

```diff
@@ -0,0 +1,273 @@
+---
+layout: default
+title:  "Kleisli"
+section: "data"
+source: "https://github.com/non/cats/blob/master/core/src/main/scala/cats/data/Kleisli.scala"
+scaladoc: "#cats.data.Kleisli"
+---
+# Kleisli
+Kleisli is a data type that will come in handy often, especially if you are working with monadic functions.
+Monadic functions are functions that return a monadic value - for instance, a function may return an
+`Option[Int]` or an `Xor[String, List[Double]]`.
+
+How then do we compose these functions together nicely? We cannot use the usual `compose` or `andThen` methods
+without having functions take an `Option` or `Xor` as a parameter, which can be strange and unwieldy.
+
+We may also have several functions which depend on some environment and want a nice way to compose these functions
+to ensure they all receive the same environment. Or perhaps we have functions which depend on their own "local"
+configuration and all the configurations together make up a "global" application configuration. How do we
+have these functions play nice with each other despite each only knowing about their own local requirements?
+
+These situations are where `Kleisli` is immensely helpful.
+
+## Functions
+One of the most useful properties of functions is that they **compose**. That is, given a function
+`A => B` and a function `B => C`, we can combine them to create a new function `A => C`. It is through
+this compositional property that we are able to write many small functions and compose them together
+to create a larger one that suits our needs.
+
```

Comment:
I think it would be useful to provide an example of composing functions here, just to be clear how that works:

```scala
val twice: Int => Int =
  x => x * 2

val countCats: Int => String =
  x => if (x == 1) "1 cat" else s"$x cats"

val twiceAsManyCats: Int => String =
  twice andThen countCats
```